src/fuzz/fuzz-journald-stream.c

```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <linux/sockios.h>
+
+#include "fd-util.h"
+#include "fuzz.h"
+#include "fuzz-journald.h"
+#include "journald-stream.h"
+
+static int stream_fds[2] = { -1, -1 };
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        Server s;
+        StdoutStream *stream;
+        int v;
+
+        if (size == 0)
+                return 0;
+
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        assert_se(socketpair(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0, stream_fds) >= 0);
+        dummy_server_init(&s, NULL, 0);
+        assert_se(stdout_stream_install(&s, stream_fds[0], &stream) >= 0);
+        assert_se(write(stream_fds[1], data, size) == (ssize_t) size);
+        while (ioctl(stream_fds[0], SIOCINQ, &v) == 0 && v)
```

Comment:
Not matter, but should this be `v > 0`?

Reply:
To judge from the code at https://github.com/torvalds/linux/blob/a978a5b8d83f795e107a2ff759b28643739be70e/net/unix/af_unix.c#L2623-L2629 `k` shouldn't be negative. If it was, it would mean that less than zero bytes could be read from the socket, which would be very weird to say the least :-) Anyway, I could add something like `assert_se(v > 0)` and a comment to make my assumptions more clear later, but, in the meantime, it would be great to merge it right away if the PR looks good to you so that ClusterFuzz would start build the fuzzer today (the build window is still open judging by https://oss-fuzz-build-logs.storage.googleapis.com/index.html).

Reply:
Yeah, let's add that later.

Reply:
What's great about the fuzzers is that they're run literally billions of times so if my assumptions are wrong, we're going to find out very soon :-)